apps/coscrad-frontend/src/components/commands/connections/tag-resource-or-note-form.tsx

```diff
@@ -0,0 +1,87 @@
+import { AggregateType } from '@coscrad/api-interfaces';
+import { isNonEmptyString } from '@coscrad/validation-constraints';
+import {
+    Box,
+    Button,
+    FormControl,
+    FormGroup,
+    MenuItem,
+    Select,
+    Stack,
+    Typography,
+} from '@mui/material';
+import { useState } from 'react';
+import { useLoadableTags } from '../../../store/slices/tagSlice/hooks/use-loadable-tags';
+import { ErrorDisplay } from '../../error-display/error-display';
+import { Loading } from '../../loading';
+import { findOriginalTextItem } from '../../notes/shared/find-original-text-item';
+
+export interface TagResourceOrNoteProps {
+    onSubmitForm: (fsa?: { type: string; payload: Record<string, unknown> }) => void;
+    bindProps: Record<string, unknown>;
+}
+
+export const TagResourceOrNoteForm = ({
+    onSubmitForm,
+    bindProps,
+}: TagResourceOrNoteProps): JSX.Element => {
+    const [selectedTagId, setSelectedTagId] = useState<string>(null);
+
+    const { data, isLoading, errorInfo } = useLoadableTags();
+
+    if (isLoading || data === null) return <Loading />;
+
+    if (errorInfo) return <ErrorDisplay {...errorInfo} />;
+
+    const isFormComplete = isNonEmptyString(selectedTagId);
+
+    const { entities: allTags } = data;
+
+    return (
+        <Box>
+            <Typography variant="h2">Tag Resource or Note</Typography>
+            <FormControl fullWidth>
+                <Stack>
+                    <FormGroup>
+                        <Select
```

Comment:
It seems like we should have been able to reuse the dynamic select if it's API were better designed. Let's circle back to this when we have to do another one of these.